src/UI/WB.UI.Headquarters.Core/Controllers/Api/PublicApi/Graphql/Maps/Map.cs

```diff
@@ -29,10 +31,24 @@ protected override void Configure(IObjectTypeDescriptor<MapBrowseItem> descripto
                 .Resolve(context =>
                 {
                     var mapId = context.Parent<MapBrowseItem>().Id;
+                    var user = context.Service<IAuthorizedUser>();
+                    var unitOfWork = context.Service<IUnitOfWork>();
 
-                    return context.Service<IUnitOfWork>().Session.Query<UserMap>()
-                        .Where(a => mapId == a.Map.Id)
-                        .ToList();
+                    var mapUsers = unitOfWork.Session.Query<UserMap>()
+                        .Where(a => mapId == a.Map.Id);
+                    
+                    if (user.IsSupervisor)
+                    {
+                        //limit by team
+                        var team = unitOfWork.Session.Query<HqUser>()
+                            .Where(x => x.WorkspaceProfile.SupervisorId == user.Id)
+                            .Select(x=> x.UserName).ToList();
```

Comment:
remove ToList, add supervisor name in query '''Where''' 
In this case all query will be executed on sql side